src/parsing.jl

```diff
@@ -304,11 +290,9 @@ function pqparse(::Type{UTCDateTime}, str::AbstractString)
         return UTCDateTime(typemin(DateTime))
     end
 
-    # Postgres should give us strings ending with +00, +00:00, -00:00
-    # We use the regex below to strip these character off before parsing, iff,
-    # the values after the `-`/`+` are `0` or `:`. This means parsing will fail if
-    # we're asked to parse a non-UTC string like +04:00.
-    return _pqparse(UTCDateTime, replace(str, r"[-|\+][0|:]*$" => ""))
+    # Postgres should always give us strings ending with +00 if our timezone is set to UTC
+    # which is the default
+    return parse(UTCDateTime, _trunc_seconds(replace(str, "+00" => "")), TIMESTAMP_FORMAT)
```

Comment:
Previously the "_trunc_seconds" version was conditional on the previous attempt failing. Is there a reason to always run it? It's doing a regex replace that we could skip if it's not triggered, which I expect is often the case (except for cases where people use `now()` for the current time, which will almost always have microseconds). 

Like we could do:
```julia
str = replace(str, "+00" => "")
parsed = tryparse(UTCDateTime, str, TIMESTAMP_FORMAT)
parsed != nothing && return parsed
return parse(UTCDateTime, _trunc_seconds(str), TIMESTAMP_FORMAT)
```

What do you think?

Reply:
Sure. That seems fine, though in that case we should also update the `DateTime` method above, so that it's consistent.

Reply:
Updated and made all datetime methods more consistent.